### ISSUE TITLE
GoAccess debian repository install does not work on Stretch (libssl1.0.2)

### ISSUE THREAD

**Issue by cedhuf** [allinurl/goaccess#716]
Hi,
Due to unmet dependency, it's not possible to install GoAccess from your repository [http://deb.goaccess.io/](http://deb.goaccess.io/) on debian stretch.
- GoAccess says it depends on : libssl1.0.0 (>= 1.0.0).
- libssl1.0.2 and libssl1.1 are the version available on stretch (and are installed).

Is something wrong on my side or do I need to wait for an update ?

Thanks.
Br,

**Comment by allinurl** [comment]
Interesting... I'm curious why `libssl1.0.0 (>= 1.0.0)` is failing to check `libssl1.0.2`. I'll look into this and see what I find and post back. 

In the mean time, installing from [**source**](https://goaccess.io/download#installation) should do it, just make sure to configure with `--with-openssl` to add SSL/TLS support.

**Comment by cedhuf** [comment]
Ok, thanks !

**Comment by spiffin** [comment]
Hello. Any progress on this? Stretch is now the stable Debian release - and it still won't install from the GoAcess Debian repo:

The following packages have unmet dependencies:
 goaccess : Depends: libssl1.0.0 (>= 1.0.0) but it is not installable
E: Unable to correct problems, you have held broken packages.

Thanks.

**Comment by allinurl** [comment]
Haven't had the chance to look into this. But thanks for the reminder, I'll take a look at it this week and post back.

**Comment by pdecat** [comment]
On jessie, libssl1.0.0 used to have an 1.0.1 version:

```
root@d19c8b7c0531:/# apt-cache policy libssl1.0.0                                                                                                                                                              
libssl1.0.0:
  Installed: (none)
  Candidate: 1.0.1t-1+deb8u6
  Version table:
     1.0.1t-1+deb8u6 0
        500 http://deb.debian.org/debian/ jessie/main amd64 Packages
        500 http://security.debian.org/ jessie/updates/main amd64 Packages
```

On stretch, libssl1.0.0 only has no candidates and libssl1.0.2 is a distinct package:

```
root@e7e3c90a9314:/# apt install goaccess
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 goaccess : Depends: libssl1.0.0 (>= 1.0.0) but it is not installable
E: Unable to correct problems, you have held broken packages.
```

```
root@e7e3c90a9314:/# apt policy libssl1.0.0
libssl1.0.0:
  Installed: (none)
  Candidate: (none)
  Version table:
```

```
root@e7e3c90a9314:/# apt policy libssl1.0.2
libssl1.0.2:
  Installed: (none)
  Candidate: 1.0.2l-2
  Version table:
     1.0.2l-2 500
        500 http://deb.debian.org/debian stretch/main amd64 Packages
```

**Comment by CPC67128** [comment]
Hi
Same issue for me (Debian 9 Stretch), I will try from source as suggested.

![2017-07-20_002033](https://user-images.githubusercontent.com/1124698/28414121-73963ec6-6d4a-11e7-8d86-4461b6c2cb54.png)

**Comment by alison-f-cooper** [comment]
I'm experiencing this problem as well, in attempting to install nginx via a docker file. 

``` nginx : Depends: libssl1.1 (>= 1.1.0) but it is not installable
E: Unable to correct problems, you have held broken packages.
```

**Comment by Nerothos** [comment]
I have the same issue on Debian 9.1.
```
sudo apt policy libssl1.0.2
libssl1.0.2:
  Installed: 1.0.2l-2
  Candidate: 1.0.2l-2
  Version table:
 *** 1.0.2l-2 500
        500 http://ftp.debian.org/debian stable/main amd64 Packages
        100 /var/lib/dpkg/status
$ sudo apt install goaccess
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 goaccess : Depends: libssl1.0.0 (>= 1.0.0) but it is not installable
E: Unable to correct problems, you have held broken packages.
```

**Comment by allinurl** [comment]
I was looking at this yesterday and noticed that even though I could package it for the right dependency (i.e., libssl1.0.2 on strech), the binary wouldn't work on stretch since it was compiled with libssl1.0.0 shared libraries. A possible solution would be to just drop libssl as dependency.

**Comment by johannsa** [comment]
Any update on this?

**Comment by allinurl** [comment]
Alright, I've built v1.2 for stretch. Please give this a shot per this [**instructions**](https://goaccess.io/download#official-repo) and let me know if that solves the issue. Thanks!

**Comment by pdecat** [comment]
@allinurl perfect, thanks!

**Comment by allinurl** [comment]
Great, thanks for posting back!

**Comment by andrem1987** [comment]
Hey,

i'm experiencing the same issue with goacces v1.3 from the official respository on Debian stretch 9.6.
The libraries `libcrypto.so.1.0.2` and `libssl.so.1.0.2` are installed properly.  When trying to run goaccess i receive the error `goaccess: error while loading shared libraries: libssl.so.1.0.0: cannot open shared object file: No such file or directory`.
I followed the official [setup-instructions](https://goaccess.io/download#official-repo).

**Comment by allinurl** [comment]
@andrem1987 Thanks for reporting this! Please give this a shot again and let me know if this time works from your end.

**Comment by andrem1987** [comment]
@allinurl It works perfect! Thank you.

**Comment by joonasvanhatapio** [comment]
This same issue happens on Ubuntu 18.04 installations using deb.goaccess.io.
Update from 1.2 to 1.3 is not available due to unmet dependencies and on new machines installation fails with the following error message. 

`Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 goaccess : Depends: libssl1.0.2 (>= 1.0.2) but it is not installable
E: Unable to correct problems, you have held broken packages.`

apt list --installed | grep libssl
libssl1.0.0/bionic-updates,bionic-security,now 1.0.2n-1ubuntu5.1 amd64 [installed]
libssl1.1/bionic,now 1.1.1-3+ubuntu18.04.1+deb.sury.org+3 amd64 [installed,automatic]

**Comment by allinurl** [comment]
@joonasvanhatapio Thanks for reporting this. Please try again:

    $ sudo apt-get update
    $ sudo apt-get install goaccess

**Comment by joonasvanhatapio** [comment]
I confirm that it's working now on Ubuntu 18.04, thank you!

**Comment by shaal** [comment]
I have the same issue on Ubuntu 18.10
I tried using the instructions here https://goaccess.io/download#official-repo
but still no success

**Comment by KeyvanJS** [comment]
> 
> 
> I get the same error on Ubuntu 18.10
> `The following packages have unmet dependencies: goaccess : Depends: libssl1.0.2 (>= 1.0.2) but it is not installable E: Unable to correct problems, you have held broken packages. `
> I tried using the instructions here https://goaccess.io/download#official-repo
> but still no success

Same issue with Ubuntu 18.10

**Comment by Niek** [comment]
Same here on Ubuntu 18.10:
```bash
# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.10
Release:	18.10
Codename:	cosmic
# apt install goaccess
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 goaccess : Depends: libssl1.0.2 (>= 1.0.2) but it is not installable
E: Unable to correct problems, you have held broken packages.
```

**Comment by MacGyver27** [comment]
same for Debian 10!!
Debian 4.19.67-2+deb10u1 (2019-09-20) x86_64 GNU/Linux
```
apt-get install goaccess
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 goaccess : Depends: libssl1.0.2 (>= 1.0.2) but it is not installable
E: Unable to correct problems, you have held broken packages.
```

**Comment by aremai** [comment]
any news regarding Debian Buster (10) ?
would love to install goaccess

**Comment by aremai** [comment]
i've been following the official guide here:
https://goaccess.io/download#official-repo

**Comment by allinurl** [comment]
v1.4 will support Buster. ETA end of March, early Apr.

**Comment by whalehub** [comment]
@allinurl Any update on this?

**Comment by allinurl** [comment]
@whalehub it's slightly delayed from the milestone release date, but I still expect to deploy pretty soon. I want to make sure I address some critical issues on the upcoming release. Please stay tuned.

**Comment by paolodina** [comment]
Sorry for bothering, do you think this can be fixed soon/next days, now that v1.4 is out?

**Comment by allinurl** [comment]
[**v1.4**](https://goaccess.io/download) is out! Available as well in the deb repo for the following distributions: 

```
goaccess |  1:1.4-xenial |  xenial | amd64, i386
goaccess |  1:1.4-bionic |  bionic | amd64
goaccess |    1:1.4-eoan |    eoan | amd64
goaccess |   1:1.4-focal |   focal | amd64
goaccess |  1:1.4-jessie |  jessie | amd64
goaccess | 1:1.4-stretch | stretch | amd64, i386
goaccess |  1:1.4-buster |  buster | amd64, i386
```
If you get a `GPG error The following signatures couldn't be verified because the public key is not available.` Please make sure to download the new public key as shown [**here**](https://goaccess.io/download#official-repo) above to correct this.
